src/services/create-project.service.js

```diff
@@ -61,11 +61,11 @@ export default (
 
   onStatusUpdate('Created parent directory');
 
-  const id = getProjectId(projectName);
+  const projectDirectoryName = getProjectNameSlug(projectName);
```

Comment:
Good catch.
Renaming the `id` and method is great as this was confusing and it has nothing to do with the `projectId` as we're using uuid now. Maybe we could add something to the docs about `projectId` handling as we're having it now - could be helpful for new contributors.

Reply:
Yeah, I added some comments to the `types.js` to explain the distinction a bit more, but maybe we could go into detail somewhere? Not sure we have a doc for explaining how our data structures are set up, and I'm not sure if it would be valuable (might be better to just direct people to `types.js` since they'll always be up-to-date)

Reply:
Yes, pointing to `types.js` is probably better.
I'll add a comment to `types.js` as I think there is a comment not correct.

Reply:
I thought the comment in [line 97](https://github.com/joshwcomeau/guppy/blob/project-configuration-modal-tweaks/src/types.js#L97) is not correct. But I think it's OK.
The uuid is only in the guppy key and at every `byId` state for newly created projects. E.g. screenshot of projects with two old projects & one newly create project:
![grafik](https://user-images.githubusercontent.com/3046542/45269106-6b16ac00-b488-11e8-95ae-0de10d9e236b.png)

Reply:
Yeah, it is a bit confusing that we have this "External" `Project` type and then the internal "what's actually in the state" type.

The reason for it is that `Project` combines a bunch of reducers into the best API for components, but maybe we could find a way to make it more consistent?

I'll add a note about legacy projects in `types`